lib/astutils.cpp

```diff
@@ -2576,9 +2580,7 @@ bool isVariableChanged(const Token *tok, int indirect, const Settings *settings,
     }
 
     const Token *parent = tok2->astParent();
-    while (Token::Match(parent, ".|::"))
```

Comment:
spontanously this while loop makes sense to me.. at least if it's rhs operand.. did you determine that the condition can't be true?

Reply:
I removed this mainly because I didn't see a use for it. Restored.